server/src/main/java/org/elasticsearch/repositories/RepositoriesStats.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.repositories;
+
+import org.elasticsearch.TransportVersion;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.xcontent.ToXContentFragment;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public class RepositoriesStats implements Writeable, ToXContentFragment {
+
+    private final Map<String, ThrottlingStats> repositoryThrottlingStats;
+
+    public RepositoriesStats(StreamInput in) throws IOException {
+        if (in.getTransportVersion().onOrAfter(TransportVersion.V_8_500_010)) {
+            repositoryThrottlingStats = in.readMap(StreamInput::readString, ThrottlingStats::new);
+        } else {
+            repositoryThrottlingStats = new HashMap<>();
+        }
+    }
+
+    public RepositoriesStats(Map<String, ThrottlingStats> repositoryThrottlingStats) {
+        this.repositoryThrottlingStats = new HashMap<>(repositoryThrottlingStats);
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        if (out.getTransportVersion().onOrAfter(TransportVersion.V_8_500_010)) {
+            out.writeMap(repositoryThrottlingStats, StreamOutput::writeString, (o, v) -> v.writeTo(o));
+        }
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.field("repository", repositoryThrottlingStats);
+        return builder;
+    }
+
+    public Map<String, ThrottlingStats> getRepositoryThrottlingStats() {
+        return Collections.unmodifiableMap(repositoryThrottlingStats);
+    }
+
+    public static class Builder {
+        private final Map<String, ThrottlingStats> repoThrottlingStats = new HashMap<>();
+
+        public Builder add(String repoName, long totalReadThrottledNanos, long totalWriteThrottledNanos) {
+            repoThrottlingStats.put(repoName, new ThrottlingStats(totalReadThrottledNanos, totalWriteThrottledNanos));
+            return this;
+        }
+
+        public RepositoriesStats build() {
+            return new RepositoriesStats(repoThrottlingStats);
+        }
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class ThrottlingStats implements ToXContentObject, Writeable {
```

Comment:
Maybe just `RepositoryStats`, we might want to add more stats later. Also can this be a `record`?

Reply:
1) Renaming is appealing, but if we add more stats in a future, we'd need to keep this and future counters separated on a semantic level anyway. So, I'd prefer to keep current naming for now. When we stats come, we'll add *Something*Stats holder right besides current ThrottlingStats.

2) Yep, record looks good here